trigger/index.ts

```diff
@@ -0,0 +1,88 @@
+// @ts-ignore
+import useSWR, {
+  Middleware,
+  SWRHook,
+  Key,
+  Fetcher,
+  SWRConfiguration,
+  SWRResponse
+} from 'swr'
+import { useCallback } from 'react'
+
+import { withMiddleware } from '../src/utils/with-middleware'
+import { serialize } from '../src/utils/serialize'
+
+export const trigger = ((<Data, Error>(useSWRNext: SWRHook) => (
+  key: Key,
+  fetcher: Fetcher<Data>,
+  config: SWRConfiguration<Data, Error>
+): SWRTriggerResponse<Data, Error> => {
+  // Disable all revalidations.
+  config.revalidateOnFocus = false
+  config.revalidateWhenStale = false
+  config.revalidateOnReconnect = false
+
+  // Disable options that are imcompatible with trigger mode.
+  config.suspense = false
+  config.shouldRetryOnError = false
+  config.refreshInterval = 0
+  config.loadingTimeout = 0
+  config.fetcher = undefined
+
+  // Extract callbacks.
+  const { onSuccess, onError } = config
+
+  // Disable fetcher.
+  const swr = useSWRNext(key, null, config) as SWRTriggerResponse
+
+  // Serialize the key and get args.
+  const [keyStr, args] = serialize(key)
+
+  // Method to trigger the request manually.
+  swr.trigger = useCallback(
+    (...extraArgs) => {
+      // If it's not ready to fetch, we throw an error.
+      if (!keyStr) throw new Error('Empty SWR key.')
+      if (!fetcher) throw new Error('Missing SWR fetcher.')
+
+      // Mutate the SWR data and return the result.
```

Comment:
What happens when a revalidation happens before the `useSWRTrigger` fetcher resolves?
Does SWR have mechanism to prevent race conditions?

Reply:
Yeah, since we use `mutate` under the hood, race conditions will be prevented. In the case you described, the revalidation result will be ignored. 

All the 3 cases: https://github.com/vercel/swr/blob/f10e744e78bca9679c104dab7d04c39fb0f06766/src/use-swr.ts#L328-L339